### ISSUE TITLE
SC2198 false positive

### ISSUE THREAD

**Issue by maxpagani** [koalaman/shellcheck#2923]
#### For bugs
- Rule Id: SC2198
- My shellcheck version: 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/usr/bin/env bash
if [ "${@: -1}" == "foo" ]
then
  echo "last arg is 'foo'"
fi
```

#### Here's what shellcheck currently says:
Arrays don't work as operands in [ ]. Use a loop (or concatenate with * instead of @).

#### Here's what I wanted or expected to see:
No error. The expression in the test is not a concatenation, but the extraction of last parameter.
